### ISSUE TITLE
The UI didn't work after I use SparkComputationGraph.

### ISSUE THREAD

**Issue by shaoweijob** [deeplearning4j/deeplearning4j#5080]
Hello， I use SparkComputationGraph to build a network with skip connection. I want to use RemoteUIStatsStorageRouter to monitor the training steps. However, the UI didn't print out expected information in the Overview such as score, learning rate, number of iterations, epoch and etc.

"http://localhost:9000/train/overview/data" shows
{"stdevActivations":{},"updateRatios":{},"scores":[],"scoresIter":[],"model":[["Model Type",""],["Layers",""],["Total Parameters",""]],"stdevGradients":{},"perf":[["Start Time",""],["Total Runtime",""],["Last Update",""],["Total Parameter Updates",""],["Updates/sec",""],["Examples/sec",""]],"updateTimestamp":1525680603431,"stdevUpdates":{}}

* Deeplearning4j 1.0 

` 
/** spark graph use **/
val sparkGraph = new SparkComputationGraph(sparkConf, networkConfig, trainingMaster)
val seesionId = "test"
val remoteUIRouter: StatsStorageRouter = new RemoteUIStatsStorageRouter("http://localhost:9000")
val statsListener = new StatsListener(null, null, null, seesionId, null)
sparkGraph.setListeners(remoteUIRouter, Collections.singletonList(statsListener))
`

`
UIServer uiServer = UIServer.getInstance();
uiServer.enableRemoteListener();

Could you kindly take a look?
Thank you very much!
![1](https://user-images.githubusercontent.com/18146142/39741881-10ce9422-52ce-11e8-91b1-6e33c2942ce8.png)
![2](https://user-images.githubusercontent.com/18146142/39741882-1118a29c-52ce-11e8-8817-d4848f3d2896.png)
![3](https://user-images.githubusercontent.com/18146142/39741884-115c94de-52ce-11e8-8643-983b68b2db3a.png)

**Comment by AlexDBlack** [comment]
Try first with the following constructor, with the frequency set to 1 - and let me know if that works:
https://github.com/deeplearning4j/deeplearning4j/blob/1f8af820c29cc5567a2c5eaa290f094c4d1492a7/deeplearning4j-ui-parent/deeplearning4j-ui-model/src/main/java/org/deeplearning4j/ui/stats/StatsListener.java#L44
Other than the use of that constructor, your code look ok at first glance.

**Comment by shaoweijob** [comment]
I'll try to do it right away. @AlexDBlack

**Comment by shaoweijob** [comment]
`val remoteUIRouter: StatsStorageRouter = new RemoteUIStatsStorageRouter("http://localhost:9000")
   val statsListener = new StatsListener(remoteUIRouter, 1)`

The program throw exception that "Task not serializable" 

Exception in thread "main" org.apache.spark.SparkException: Task not serializable
        at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:298)
        at org.apache.spark.util.ClosureCleaner$.org$apache$spark$util$ClosureCleaner$$clean(ClosureCleaner.scala:288)
        at org.apache.spark.util.ClosureCleaner$.clean(ClosureCleaner.scala:108)
        at org.apache.spark.SparkContext.clean(SparkContext.scala:2287)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1.apply(RDD.scala:794)
        at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1.apply(RDD.scala:793)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
        at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
        at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
        at org.apache.spark.rdd.RDD.mapPartitions(RDD.scala:793)
        at org.apache.spark.api.java.JavaRDDLike$class.mapPartitions(JavaRDDLike.scala:155)
        at org.apache.spark.api.java.AbstractJavaRDDLike.mapPartitions(JavaRDDLike.scala:45)
        at org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingMaster.doIterationPathsMDS(ParameterAveragingTrainingMaster.java:738)
        at org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingMaster.executeTrainingPathsMDSHelper(ParameterAveragingTrainingMaster.java:592)
        at org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingMaster.executeTrainingMDS(ParameterAveragingTrainingMaster.java:459)
        at org.deeplearning4j.spark.impl.graph.SparkComputationGraph.fitMultiDataSet(SparkComputationGraph.java:235)
        at org.deeplearning4j.spark.earlystopping.SparkEarlyStoppingGraphTrainer.fitMulti(SparkEarlyStoppingGraphTrainer.java:81)
        at org.deeplearning4j.spark.earlystopping.BaseSparkEarlyStoppingTrainer.fit(BaseSparkEarlyStoppingTrainer.java:115)
        at com.finup.roboter.core.model.DnnParameter.DL4JNetworkUtil.fitEarlyStop(DL4JNetworkUtil.scala:299)
        at com.finup.roboter.core.model.DnnParameter.Dl4jMultiModel.fit(Dl4jMultiModel.scala:290)
        at com.finup.models.dl4j.SnaContactModel$.main(SnaContactModel.scala:153)
        at com.finup.models.dl4j.SnaContactModel.main(SnaContactModel.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:755)
        at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:180)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:205)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:119)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.io.NotSerializableException: org.deeplearning4j.api.storage.impl.RemoteUIStatsStorageRouter
Serialization stack:
        - object not serializable (class: org.deeplearning4j.api.storage.impl.RemoteUIStatsStorageRouter, value: org.deeplearning4j.api.storage.impl.RemoteUIStatsStorageRouter@5396a0b2)
        - field (class: org.deeplearning4j.ui.stats.BaseStatsListener, name: router, type: interface org.deeplearning4j.api.storage.StatsStorageRouter)
        - object (class org.deeplearning4j.ui.stats.StatsListener, org.deeplearning4j.ui.stats.StatsListener@4b0cb52)
        - writeObject data (class: java.util.ArrayList)
        - object (class java.util.ArrayList, [org.deeplearning4j.ui.stats.StatsListener@4b0cb52])
        - field (class: org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker, name: iterationListeners, type: interface java.util.Collection)
        - object (class org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker, org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker@5d7061a1)
        - field (class: org.deeplearning4j.spark.api.worker.ExecuteWorkerMultiDataSetFlatMapAdapter, name: worker, type: interface org.deeplearning4j.spark.api.TrainingWorker)
        - object (class org.deeplearning4j.spark.api.worker.ExecuteWorkerMultiDataSetFlatMapAdapter, org.deeplearning4j.spark.api.worker.ExecuteWorkerMultiDataSetFlatMapAdapter@50edd335)
        - field (class: org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMapAdapter, name: workerFlatMap, type: interface org.datavec.spark.functions.FlatMapFunctionAdapter)
        - object (class org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMapAdapter, org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMapAdapter@33f0d842)
        - field (class: org.datavec.spark.transform.BaseFlatMapFunctionAdaptee, name: adapter, type: interface org.datavec.spark.functions.FlatMapFunctionAdapter)
        - object (class org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMap, org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMap@305fb12d)
        - field (class: org.apache.spark.api.java.JavaRDDLike$$anonfun$fn$4$1, name: f$6, type: interface org.apache.spark.api.java.function.FlatMapFunction)
        - object (class org.apache.spark.api.java.JavaRDDLike$$anonfun$fn$4$1, <function1>)
        at org.apache.spark.serializer.SerializationDebugger$.improveException(SerializationDebugger.scala:40)
        at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:46)
        at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
        at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:295)
        ... 30 more  @AlexDBlack

**Comment by shaoweijob** [comment]
I used this code to solve this problem
`
val statsConfig = new DefaultStatsUpdateConfiguration.Builder().reportingFrequency(1).build()
    val statsListener = new StatsListener(null, null, statsConfig, seesionId, null)
`

Thank you for your appointment very much. @AlexDBlack
